### PR TITLE
feat: provides interoperability with Selenium 4

### LINKED ISSUE #272: Interoperability with Selenium 4
It'd be great to make some changes in codebase in order to support Selenium 4. I've started doing that. You can see the progress [here](https://github.com/mabartos/arquillian-extension-drone/tree/selenium4).

### PR DESCRIPTION
**Pull request by mabartos**
This PR provides support for upcoming Selenium 4. At this moment, the most recent released version of Selenium is 4.0.0-RC-2. This PR prepares for full support of Selenium 4. After releasing GA, it's necessary to do some changes in pom files regarding versions properties. Moreover, _htmlUnitDriver_ will be released soon with Selenium 4 compatibility.

There are some failing tests for WebDriver caused by PhantomJS incompatibility with Selenium 4 and some issues with TLS certificate in GitHub tests.

Initial support for #272

### CODE REVIEW

**Comment by bartoszmajsak** [comment]
Thank you for your contribution. It seems that I have to dust off the CI job (or migrate it to something else) as it's not giving back any feedback.

**Comment by bartoszmajsak** [comment]
@mabartos please update the branch to have the latest `CircleCI` build enabled. It works right now.

I ran your branch locally and I'm facing following errors:

```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/bartek/code/redhat/arquillian/arquillian-extension-drone/drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/htmlunit/DroneHtmlUnitDriver.java:[7,8] cannot access org.openqa.selenium.internal.FindsById
  class file for org.openqa.selenium.internal.FindsById not found
[ERROR] /home/bartek/code/redhat/arquillian/arquillian-extension-drone/drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/htmlunit/DroneHtmlUnitDriver.java:[13,5] method does not override or implement a method from a supertype
[ERROR] /home/bartek/code/redhat/arquillian/arquillian-extension-drone/drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/htmlunit/DroneHtmlUnitDriver.java:[20,5] method does not override or implement a method from a supertype
[ERROR] /home/bartek/code/redhat/arquillian/arquillian-extension-drone/drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/htmlunit/DroneHtmlUnitDriver.java:[22,16] cannot find symbol
  symbol:   variable super
  location: class org.jboss.arquillian.drone.webdriver.htmlunit.DroneHtmlUnitDriver
[ERROR] /home/bartek/code/redhat/arquillian/arquillian-extension-drone/drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/factory/HtmlUnitDriverFactory.java:[39,69] type argument org.jboss.arquillian.drone.webdriver.htmlunit.DroneHtmlUnitDriver is not within bounds of type-variable T
[ERROR] /home/bartek/code/redhat/arquillian/arquillian-extension-drone/drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/factory/HtmlUnitDriverFactory.java:[63,17] cannot find symbol
  symbol:   method quit()
  location: variable instance of type org.jboss.arquillian.drone.webdriver.htmlunit.DroneHtmlUnitDriver
```

That's my env:

```
Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-04T21:00:29+02:00)
Maven home: /home/bartek/.m2/wrapper/dists/apache-maven-3.6.1-bin/38pn40mp89t5c94bjdbeod370m/apache-maven-3.6.1
Java version: 1.8.0_181, vendor: Oracle Corporation, runtime: /usr/java/jdk1.8.0_181/jre
Default locale: en_GB, platform encoding: UTF-8
OS name: "linux", version: "5.14.13-100.fc33.x86_64", arch: "amd64", family: "unix"
```

**Comment by mabartos** [comment]
@bartoszmajsak Oh, I just found out we need to wait until the HtmlUnit driver with Selenium 4 support is released. Sorry, I should have found out earlier.

**Comment by bartoszmajsak** [comment]
> @bartoszmajsak Oh, I just found out we need to wait until the HtmlUnit driver with Selenium 4 support is released. Sorry, I should have found out earlier.

No worries, let's wait for the fix for your issue reported upstream https://github.com/SeleniumHQ/htmlunit-driver/issues/105.

**Comment by bartoszmajsak** [comment]
@mabartos we have circleci jobs for multiple browsers now.

**Comment by bartoszmajsak** [comment]
@mabartos I tried to update your branch, but somewhat I don't see the new jobs kicked in. Can you take care of it?

**Comment by bartoszmajsak** [comment]
@mabartos all builds are failing b/c of HtmlUnit incompatible code.

**Comment by mabartos** [comment]
@bartoszmajsak It seems the failures are caused by codeborne PhantomJS driver, for which the Selenium 4 is not supported yet. I created PR with Selenium 4 support. See [here](https://github.com/codeborne/ghostdriver/pull/6).

**Comment by bartoszmajsak** [comment]
> @bartoszmajsak It seems the failures are caused by codeborne PhantomJS driver, for which the Selenium 4 is not supported yet. I created PR with Selenium 4 support. See [here](https://github.com/codeborne/ghostdriver/pull/6).

Oh ok that too :)

**Comment by bartoszmajsak** [comment]
@mabartos we got ourselves a bit into snowball situation here. Now both Firefox and Edge are failing because of not being able to find matching capabilities. Which might, in fact, indicate that both versions are not yet compatible with Selenium 4.0?

**Comment by bartoszmajsak** [comment]
It could be that we are facing this issue https://github.com/mozilla/geckodriver/issues/1844 with failing firefox build. We are using geckodriver `0.30.0` under the hood but it still seems to have default `user.js` preferences in the temporary folder without our changes.

**Comment by bartoszmajsak** [comment]
There are quite a few things that are not great right now. First thing is that underneath we are using `selenium-standalone-server-3.11.0`. In order to start using `4.x` we have to swap implementation to fetch releases from GitHub. This itself is not a big deal, but it has a few consequences:

* we have to change the way how we invoke the server - there's a new subcommand `standalone` now. In consequence, we break backward compatibility unless with patch handling for older versions
* we will need to redo quite a few tests 
* custom profile for Firefox is still broken :)
* in general documentation need some love

I will bring these changes to master in order to release `3.0.0-alpha-1` and move things forward, but we have to take care of it sooner than later.

This PR needs to wait until https://github.com/arquillian/arquillian-extension-drone/issues/296 is resolved.

File: drone-bom/pom.xml [hunk 1]
```diff
@@ -39,9 +39,10 @@
   </developers>
 
   <properties>
-    <version.org.jboss.arquillian.selenium>3.11.0</version.org.jboss.arquillian.selenium>
-    <version.htmlunit.driver>2.28</version.htmlunit.driver>
-    <phantomjs.driver.version>1.4.4</phantomjs.driver.version>
+    <version.selenium>4.0.0</version.selenium>
+    <version.selenium.server>4.0.0-alpha-2</version.selenium.server>
```
Comment (bartoszmajsak) [inline drone-bom/pom.xml:43]: This is the last release when they released `selenium-server-standalone.jar`

Moving forward (`4.1.0` at this point) there's a single command which has `standalone` subcommand to use. See https://github.com/arquillian/arquillian-extension-drone/pull/273#issuecomment-977198261

File: drone-bom/pom.xml [hunk 2]
```diff
@@ -39,9 +39,10 @@
   </developers>
 
   <properties>
-    <version.org.jboss.arquillian.selenium>3.11.0</version.org.jboss.arquillian.selenium>
-    <version.htmlunit.driver>2.28</version.htmlunit.driver>
-    <phantomjs.driver.version>1.4.4</phantomjs.driver.version>
+    <version.selenium>4.0.0</version.selenium>
```
Comment (bartoszmajsak) [inline drone-bom/pom.xml:42]:
```suggestion
    <version.selenium>4.1.0</version.selenium>
```

File: drone-webdriver/pom.xml [hunk 3]
```diff
@@ -27,6 +27,7 @@
   <properties>
     <!-- m2e settings -->
     <m2e.jpa.activation>false</m2e.jpa.activation>
+    <google.code.gson.version>2.8.8</google.code.gson.version>
```
Comment (bartoszmajsak) [inline drone-webdriver/pom.xml:30]:
```suggestion
    <google.code.gson.version>2.8.9</google.code.gson.version>
```

File: pom.xml [hunk 4]
```diff
@@ -37,22 +37,23 @@
   <properties>
     <!-- Versioning -->
     <version.arquillian.core>1.6.0.Final</version.arquillian.core>
-    <version.selenium>3.11.0</version.selenium>
-    <version.htmlunit.driver>2.28</version.htmlunit.driver>
-    <phantomjs.driver.version>1.4.4</phantomjs.driver.version>
-    <version.appium.java.client>5.0.0-BETA9</version.appium.java.client>
+    <version.selenium>4.0.0</version.selenium>
```
Comment (bartoszmajsak) [inline pom.xml:40]:
```suggestion
    <version.selenium>4.1.0</version.selenium>
```

**Review by bartoszmajsak — APPROVED** [review]

**Pull request merged**